### PR TITLE
feat(layouts/shortcodes/mermaid.html-layouts/_default/baseof.html): A…

### LINKED ISSUE #594: Mermaid support
Is there any plan/way to support [mermaid](https://mermaid-js.github.io/mermaid/#/) in this theme?

### PR DESCRIPTION
**Pull request by jneo8**
Add mermaid js support for theme

### Prerequisites

Put an `x` into the box(es) that apply:

- [ ] This pull request fixes a bug.
- [x] This pull request adds a feature.
- [ ] This pull request introduces breaking change.

### Description

Add mermaid JS support for coder theme

### Issues Resolved

#594 

### Checklist

Put an `x` into the box(es) that apply:

#### General

- [x] Describe what changes are being made
- [x] Explain why and how the changes were necessary and implemented respectively
- [x] Reference issue with `#<ISSUE_NO>` if applicable

#### Resources

- [ ] If you have changed any SCSS code, run `make release` to regenerate all CSS files

#### Contributors

- [x] Add yourself to `CONTRIBUTORS.md` if you aren't on it already

### CODE REVIEW

**Comment by luizdepra** [comment]
Could please add the `mermaid` config into `exampleSite/config.toml` and a post example using the shortcode?

**Comment by gytisrepecka** [comment]
So this Pull Request was not finished?

It would be very nice to have Mermaid.js diagrams in this theme 👍

Also it would be good to provide an option to hotlink (as it is now) library's JavaScript from CDN or self-host it for GDPR reasons.

**Comment by gytisrepecka** [comment]
> question: Why we need to add mermaid to `config.toml`?

Probably to enable or disabling (inclusion of Mermaid JavaScript).

**Comment by luizdepra** [comment]
> Probably to enable or disabling (inclusion of Mermaid JavaScript).

But maybe we could include it automatically only in the pages that has the mermaid shortcode. Let me check it.

File: layouts/_default/baseof.html [hunk 1]
```diff
@@ -89,6 +89,11 @@
         integrity="sha384-5f4X0lBluNY/Ib4VhGx0Pf6iDCF99VGXJIyYy7dDLY5QlEd7Ap0hICSSZA1XYbc4" crossorigin="anonymous"></script>
     {{ end }}
 
+    {{ if (.Params.mermaid) }}
```
Comment (luizdepra) [inline layouts/_default/baseof.html:92]: Found it.

```suggestion
    {{ if .HasShortcode "mermaid" }}
```

File: layouts/_default/baseof.html [hunk 2]
```diff
@@ -89,6 +89,11 @@
...
+      <!-- MermaidJS support -->
+      <script src="https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.min.js"></script>
```
Comment (luizdepra) [inline layouts/_default/baseof.html:94]: It is a good practice to pin the version of our depedencies. So, maybe use `https://cdn.jsdelivr.net/npm/mermaid@9.1.3/dist/mermaid.core.min.js` here.

**Comment by luizdepra** [comment]
You will need to update your branch also.

Sorry for taking too long. I forgot about this PR.

File: layouts/_default/baseof.html (hunk 2 above)
Comment (jneo8) [reply layouts/_default/baseof.html]: ack

**Comment by jneo8** [comment]
@luizdepra Thanks for helping review. Updated.

File: CONTRIBUTORS.md [hunk 3]
```diff
@@ -88,6 +88,7 @@
 - [Arafat Hasan](https://github.com/arafat-hasan)
 - [YUJI](https://yuji.ne.jp/)
 - [JaeSang Yoo](https://github.com/JSYoo5B)
+<<<<<<< HEAD
```
Comment (luizdepra) [inline CONTRIBUTORS.md:91]:
```suggestion
```

Oops! xD

**Comment by luizdepra** [comment]
Nice one. Thank you.

**Pull request merged**